scan_processor_node:
  ros__parameters:
    # 스캔 데이터 필터링 파라미터
    scan_range_min: 0.0            # (m) 레이저 스캔 데이터에서 고려할 최소 거리
    scan_range_max: 10.0           # (m) 레이저 스캔 데이터에서 고려할 최대 거리
    scan_angle_min: -1.5708        # (rad) 스캔 데이터에서 고려할 최소 각도 (-π/2 ≒ -1.5708 rad)
    scan_angle_max: 1.5708         # (rad) 스캔 데이터에서 고려할 최대 각도 (π/2 ≒ 1.5708 rad)

    # DBSCAN 클러스터링 관련 파라미터
    min_cluster_points: 3          # (개) DBSCAN 클러스터링 시 하나의 클러스터로 인식하기 위한 최소 점 개수
    dbscan_epsilon: 0.02           # (m) 두 점 사이의 최대 거리 기준 (eps 값)

    # 벽 필터링 관련 파라미터
    wall_distance_threshold: 0.15   # (m) 연속된 스캔 점 사이의 최대 거리 – 이 값 이하이면 같은 그룹으로 묶음
    wall_line_max_error: 0.05      # (m) 그룹 내 점들을 직선으로 근사할 때 허용하는 최대 오차
    min_wall_cluster_points: 5     # (개) 벽으로 판단하기 위한 그룹 내 최소 점 개수
    wall_length_threshold: 0.35    # (m) 그룹 선분 길이가 이 값보다 크면 벽으로 간주

    # 멀리 있는 장애물 필터링 관련 파라미터
    far_obstacle_distance_threshold: 5.0  # (m) 센서로부터의 거리가 이 값 이상이면 '멀리 있는 장애물'로 판단
    far_obstacle_min_points: 10            # (개) 멀리 있는 장애물로 인식하기 위한 최소 스캔 점 개수
    min_obstacle_cluster_density: 10.0     # (개/m) 먼 거리에 찍힌 밀도가 낮은 클러스터는 벽으로 간주함. 
    dynamic_wall_gap_factor: 2.8           # (가중치) 먼 거리 벽 그룹화를 위한 동적 임계값 배율
    
obstacle_tracker_node:
  ros__parameters:
    merge_range_threshold: 0.1          # (m) 박스 근사 시, 마커 간 최대 거리 임계값
    smoothing_factor: 0.7              # (비율) 박스, 값이 클 수록 동일한 헤딩에 대해서 가중치 부여
    misssed_scan_threshold: 5
    heading_smoothing_factor: 0.8
    heading_outlier_threshold: 0.2
    initial_heading_damping: 0.25
    odom_output_smoothing_factor: 0.6
    heading_blend_threshold: 0.3
    heading_distance_threshold: 0.8     # (m) 거리가 이 값보다 멀면 이전 값과의 연결을 끊는 기준
    my_odom_correction_weight: 0.9      # (비율) 내 odom값과 얼마나 강하게 보정할 것인지

occupancy_grid_node:
  ros__parameters:
    pgm_file: "/rosbag/modified_new_factory.pgm"
    yaml_file: "/rosbag/modified_new_factory.yaml"
    inflation_radius: 0.15
